src/conn/conn_open.c

```diff
@@ -182,6 +167,13 @@ __wt_connection_close(WT_CONNECTION_IMPL *conn)
 		WT_TRET(__wt_dlclose(session, dlh));
 	}
 
+	/* Close the lock file, opening up the database to other connections. */
+	if (conn->lock_fh != NULL)
```

Comment:
@agorrod
> Are you sure it's safe to unlock the connection before we have finished closing? I could believe that it's safe after all other handles are closed and we ensure they are never opened again.

I think it's safe.

Previously, we:
1. Walked any remaining open files and closed them (skipping the lock file)
2. Closed some other subsystems (cache, txn, extensions)
3. Cleaned up the sessions
4. Closed the lock file

Now, we:
1. Close some other subsystems (cache, txn, extensions)
2. Close the lock file
3. Walk any remaining open files and close them
4. Clean up the sessions.

So, I'm closing the lock file earlier, but after everything has been shut down, I think that's a neutral change.

I'm closing "any remaining open files" later, after everything has been shut down. I think that's a good change, I'd rather close all the subsystems first, before checking for left-open handles.